heat/examples/vm_two_nics_with_share.yaml

```diff
@@ -76,7 +76,7 @@ resources:
             flavor: { get_param: user_vm_flavor }
             user_data: |
                 #!/usr/bin/env python
```

Comment:
this is not ok
code should be kept in it's own files/scripts (so we can unittest or parse the), and incided here using the `{get_file: some_file_name}` directive
so move the script to a script, and use the `get_file` to include it (it also allows reuse in other templates)

Reply:
I agree with you on the idea to separate the code of the Python script from the heat template. In principle, when you create a stack with Horizon using a template, you can always include an environment file in  a second field, and this how - I think - you have to pass on the Python script, but when you do so, you've got the following error message: "The value of get_file is mount_share.py inside the template. When launching a stack from this interface, the value must start with "http://" or "https://". Having google around a bit, I have found that: "This is a limitation of Horizon - it can't access local files on your machine the way the CLI client can. The recommended way to launch stacks with nested templates using Horizon is to first upload them to Swift, and then reference by URL."

Reply:
that's a pity. but we need to find other ways to achieve this. this is too error prone
(and the script better adds the export to fstab)

@alvarosimon what options do we have to get eg some of our own python code in all the VMs? cloudinit?

Reply:
Mmm.. it works for me. 

@StephaneGerardVUB have you tried this? I did a very stupid example with EB and also with some github public script files and it seems that is working, something like:

```
          user_data:
              get_file: https://raw.githubusercontent.com/alvarosimon/vsc_user_docs/master/deploy.sh
```

The only issue is the size of the file, it should be a few kilobytes because the file is injected into the metadata (but we can increase this limit if we want).

So I think that even for the tiny script @stdweird is right and we should include those scripts into a different directory in this repo and use them directly from our heat templates.

Reply:
Doing some tests but for a reason I don't understand, I can't find the export location path in the metadata anymore...